doc/api/dns.markdown

```diff
@@ -80,10 +84,13 @@ through `hints` by logically `OR`ing their values. An example usage of
 See [supported `getaddrinfo` flags](#dns_supported_getaddrinfo_flags) below for
 more information on supported flags.
 
-The callback has arguments `(err, address, family)`.  The `address` argument
-is a string representation of a IP v4 or v6 address. The `family` argument
-is either the integer 4 or 6 and denotes the family of `address` (not
-necessarily the value initially passed to `lookup`).
+The callback has arguments `(err, address, family)`. With the `all` option set,
+the arguments change to `(err, addresses)`, with `addresses` being an array of
+objects with the properties `address` and `family`.
+
+`address` is a string representation of a IP v4 or v6 address. `family` is
```

Comment:
Yes. I think it's helpful to define the default behavior before going into what it means for `all` to be set.